porcelain/api.go

```diff
@@ -55,6 +55,11 @@ func (a *API) CreatePayments(ctx context.Context, config CreatePaymentsParams) (
 	return CreatePayments(ctx, a, config)
 }
 
+// ValidateStoragePaymentCondition validates that the given condition is a payment condition and has the right values
+func (a *API) ValidateStoragePaymentCondition(ctx context.Context, condition *types.Predicate, minerAddr address.Address, commP types.CommP, pieceSize *types.BytesAmount) error {
+	return ValidateStoragePaymentCondition(ctx, condition, minerAddr, commP, pieceSize)
+}
```

Comment:
I don't see why this is in porcelain since it's just a free function, with no dependency on plumbing or the `a` receiver. I think it should be moved to `protocol/storage`. It's exposed unnecessarily widely here.

Reply:
I was hoping to keep the payment creation logic and the payment validation logic co-located. It also reduces clutter in the overcrowded client file. It's not inconceivable, that you'd want this type of validation in the form of command.

All that said, I agree it's not very porcelain-y and it doesn't really need to be here. I'm ambivalent. I'll be happy to move it in a quick follow-on PR if you still think it doesn't belong here.